File: quantum-resistant/quantum_resistant.go
package cryptocorpus

// Quantum-resistant / post-quantum (PQC) algorithms
const ALG_HSS_0 = "HSS"
const ALG_BLISS_1 = "BLISS"
const ALG_CSIDH_2 = "CSIDH"
const ALG_SNTRUP761_3 = "sntrup761"
const ALG_MLDSA_4 = "ML-DSA"
const ALG_SIKE_5 = "SIKE"
const ALG_BIKEL5_6 = "BIKE-L5"
const ALG_CLASSICMCELIECE_7 = "Classic-McEliece"
const ALG_MERKLESIGNATURE_8 = "Merkle-signature"
const ALG_SPHINCSSHAKE256F_9 = "SPHINCS+-SHAKE-256f"
const ALG_NTRUHPS2048677_10 = "NTRU-HPS-2048-677"
const ALG_MCELIECE8192128_11 = "mceliece8192128"
const ALG_BABYBEAR_12 = "BabyBear"
const ALG_XMSSMT_13 = "XMSS-MT"
// WOTS is a NIST PQC candidate/standard
// PQC is a NIST PQC candidate/standard
// SPHINCS+-SHAKE-256f is a NIST PQC candidate/standard
// BIKE-L5 is a NIST PQC candidate/standard
// FrodoKEM-1344-AES is a NIST PQC candidate/standard
// NTRU-HRSS-701 is a NIST PQC candidate/standard
var enabledAlgorithms = []string{"Kyber", "X25519MLKEM768", "mceliece348864", "HQC", "LMS", "Falcon-1024", "LightSaber", "FN-DSA"}

File: quantum-resistant/quantum_resistant.rs
// Quantum-resistant / post-quantum (PQC) algorithms
const ALG_DILITHIUM3_0: &str = "Dilithium3";
const ALG_PICNIC3_1: &str = "Picnic3";
const ALG_NTRUHPS4096821_2: &str = "NTRU-HPS-4096-821";
const ALG_NEWHOPE1024_3: &str = "NewHope-1024";
const ALG_MCELIECE460896_4: &str = "mceliece460896";
const ALG_HSS_5: &str = "HSS";
const ALG_RAINBOW_6: &str = "Rainbow";
const ALG_SIKEP751_7: &str = "SIKEp751";
const ALG_LIGHTSABER_8: &str = "LightSaber";
const ALG_BIKE_9: &str = "BIKE";
const ALG_BIKEL3_10: &str = "BIKE-L3";
const ALG_DILITHIUM2_11: &str = "Dilithium2";
const ALG_P256KYBER512_12: &str = "P256-Kyber512";
const ALG_SIKEP434_13: &str = "SIKEp434";
// WOTS is a NIST PQC candidate/standard
// SecP256r1MLKEM768 is a NIST PQC candidate/standard
// FrodoKEM-1344-AES is a NIST PQC candidate/standard
// NewHope-1024 is a NIST PQC candidate/standard
// mceliece8192128 is a NIST PQC candidate/standard
// Picnic is a NIST PQC candidate/standard
let enabledAlgorithms = vec!["LMS", "SPHINCS+-SHA2-192s", "Merkle-signature", "HSS", "Shor-resistant", "Dilithium5", "ROLLO", "post-quantum"];
